spec/yearmonth.html

```diff
@@ -3,5 +3,511 @@
+        The following steps are taken:
+      </p>
+      <emu-alg>
+        1. Let _yearMonth_ be the *this* value.
+        1. If Type(_yearMonth_) is not Object or _yearMonth_ does not have an [[InitializedTemporalYearMonth]] internal slot, then
+          1. Throw a *TypeError* exception.
+        1. Let _d_ be ? ToInteger(_day_).
+        1. If _disambiguation_ is not *undefined*, then
+          1. Set _disambiguation_ to ? ToString(_disambiguation_).
+        1. Else
+          1. Set _disambiguation_ to `"constrain"`.
+        1. Let _result_ be ? RegulateDate(_yearMonth_.[[Year]], _yearMonth_.[[Month]], _d_, _disambiguation_).
+        1. Return ? CreateDate(_result_.[[Year]], _result_.[[Month]], _result_.[[Day]]).
+      </emu-alg>
+    </emu-clause>
+  </emu-clause>
+
+  <emu-clause id=sec-temporal-yearmonth-abstract-ops>
+    <h1>Abstract operations</h1>
+
+    <emu-clause id=sec-temporal-ToPartialYearMonth aoid=ToPartialYearMonth>
+      <h1>ToPartialYearMonth ( _yearMonth_ )</h1>
+      <emu-alg>
+        1. If Type(_yearMonth_) is String, then
+          1. Return ? ParseYearMonthString(_yearMonth_).
+        1. If Type(_yearMonth_) is Object, then
+          1. Let _result_ be the new Record {
+            [[Year]]: *undefined*,
+            [[Month]]: *undefined*,
+            }.
+          1. Let _any_ be *false*.
+          1. For each row of <emu-xref href="#table-temporal-yearmonthlike-properties"></emu-xref>, except the header row, in table order, do
+            1. Let _property_ be the Property value of the current row.
+            1. Let _value_ be ? Get(_yearMonth_, _property_).
+            1. If _value_ is not *undefined*, then
+              1. Set _any_ to *true*.
+              1. Set _value_ to ? ToInteger(_value_).
+              1. Set _result_'s internal slot whose name is the Internal Slot value of the current row to _value_.
+          1. If _any_ is *false*, then
+            1. Throw a *TypeError* exception.
+          1. Return _result_.
+        1. Throw a *TypeError* exception.
+      </emu-alg>
+
+      <emu-table id="table-temporal-yearmonthlike-properties">
+        <emu-caption>Properties of a YearMonthLike</emu-caption>
+        <table class="real-table">
+          <thead>
+            <tr>
+              <th>Internal Slot</th>
+              <th>Property</th>
+            </tr>
+          </thead>
+          <tbody>
+            <tr>
+              <td>[[Month]]</td>
+              <td>`"month"`</td>
+            </tr>
+            <tr>
+              <td>[[Year]]</td>
+              <td>`"year"`</td>
+            </tr>
+          </tbody>
+        </table>
+      </emu-table>
+    </emu-clause>
+    <emu-clause id=sec-temporal-RegulateYearMonth aoid=RegulateYearMonth>
+      <h1>RegulateYearMonth ( _year_, _month_, _disambiguation_ )</h1>
+      <emu-alg>
+        1. Assert: _disambiguation_ is one of `"constrain"`, `"balance"`, or `"reject"`.
```

Comment:
this assertion is invalid - it's called with an arbitrary string value supplied by the user.

It seems like what you want is an assertion that it's a string, and then to throw a TypeError if it's not one of the allowed values.

Reply:
I'll handle this separately across the entire spec.